src/backend/dx11/src/lib.rs

```diff
@@ -415,14 +591,58 @@ impl hal::Device<Backend> for Device {
         IR: IntoIterator,
         IR::Item: Borrow<(pso::ShaderStageFlags, Range<u32>)>,
     {
-        unimplemented!()
+        // TODO: pipelinelayout
+        
+        PipelineLayout
     }
 
     fn create_graphics_pipeline<'a>(
         &self,
         desc: &pso::GraphicsPipelineDesc<'a, Backend>,
     ) -> Result<GraphicsPipeline, pso::CreationError> {
-        unimplemented!()
+        let build_shader =
+            |stage: pso::Stage, source: Option<&pso::EntryPoint<'a, Backend>>| {
+                let source = match source {
+                    Some(src) => src,
+                    None => return Ok(None),
+                };
+
+                Self::extract_entry_point(stage, source, desc.layout)
+                    .map_err(|err| pso::CreationError::Shader(err))
+            };
+
+        let vs = build_shader(pso::Stage::Vertex, Some(&desc.shaders.vertex))?.unwrap();
+        let ps = build_shader(pso::Stage::Fragment, desc.shaders.fragment.as_ref())?;
+        // TODO:
+        /*let gs = build_shader(pso::Stage::Geometry, desc.shaders.geometry.as_ref())?;
+        let ds = build_shader(pso::Stage::Domain, desc.shaders.domain.as_ref())?;
+        let hs = build_shader(pso::Stage::Hull, desc.shaders.hull.as_ref())?;*/
+
+        let (topology, input_layout) = self.create_input_layout(vs.clone(), &desc.vertex_buffers, &desc.attributes, &desc.input_assembler)?;
+        let rasterizer_state = self.create_rasterizer_state(&desc.rasterizer)?;
+        let blend_state = self.create_blend_state(&desc.blender)?;
+        let depth_stencil_state = if let Some(desc) = desc.depth_stencil {
+            Some(self.create_depth_stencil_state(&desc)?)
+        } else {
+            None
+        };
+
+        let vs = self.create_vertex_shader(vs)?;
+        let ps = if let Some(blob) = ps {
```

Comment:
`ps.map(..)`?

Reply:
IIRC compiler complained about `?` inside closure or something similar, i'll give it another try